toolchain/check/testdata/function/generic/cross_library_name_collision_public.carbon

```diff
@@ -0,0 +1,567 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/function/generic/cross_library_name_collision_public.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/function/generic/cross_library_name_collision_public.carbon
+
+// --- lib1.carbon
+
+library "[[@TEST_NAME]]";
+
+fn F[T:! type](a: T, b: T) -> T {
+  return a;
+}
+
+fn Lib1CallF[T:! type](a: T, b: T) -> T {
+  return F(a, b);
+}
+
+// --- lib2.carbon
+
+library "[[@TEST_NAME]]";
+
+// Duplicate function name in different files.
+// This should be an error due to name conflict.
+fn F[T:! type](a: T, b: T) -> T {
+  return b;
+}
+
+fn Lib2CallF[T:! type](a: T, b: T) -> T {
+  return F(a, b);
+}
+
+// --- fail_use.carbon
+
+// TODO: If the calls to `Lib1CallF` and `Lib2CallF` are in different libraries,
+// we will need a mechanism to detect this, presumably at link time.
+
+// CHECK:STDERR: fail_use.carbon:[[@LINE+9]]:1: in import [InImport]
+// CHECK:STDERR: lib2.carbon:6:1: error: duplicate name `F` being declared in the same scope [NameDeclDuplicate]
+// CHECK:STDERR: fn F[T:! type](a: T, b: T) -> T {
+// CHECK:STDERR: ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// CHECK:STDERR: fail_use.carbon:[[@LINE+5]]:1: in import [InImport]
+// CHECK:STDERR: lib1.carbon:4:1: note: name is previously declared here [NameDeclPrevious]
+// CHECK:STDERR: fn F[T:! type](a: T, b: T) -> T {
+// CHECK:STDERR: ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// CHECK:STDERR:
```

Comment:
To note, I think this is very similar to some other tests:

- https://github.com/carbon-language/carbon-lang/blob/trunk/toolchain/check/testdata/packages/no_prelude/cross_package_import.carbon#L102
- https://github.com/carbon-language/carbon-lang/blob/trunk/toolchain/check/testdata/function/declaration/import.carbon#L192

It may be redundant.

Reply:
Yeah, good point, I think it is. Deleted.